doozer/doozerlib/backend/rebaser.py

```diff
@@ -463,32 +463,30 @@ class MergeSourceResult:
             source_dockerfile_content = source_dockerfile.read()
             distgit_dockerfile.write(source_dockerfile_content)
 
-        append_gomod_patch = self._runtime.group_config.konflux.cachi2.gomod_version_patch
-        if append_gomod_patch and append_gomod_patch is not Missing:
-            gomod_path = dest_dir.joinpath('go.mod')
-            if gomod_path.exists():
-                # Read the gomod contents
-                new_lines = []
-                with open(gomod_path, "r") as file:
-                    lines = file.readlines()
-                    for line in lines:
-                        stripped_line = line.strip()
-                        match = re.match(r"(^go \d\.\d+$)", stripped_line)
-                        if match:
-                            # Append a .0 to the go mod version, if it exists
-                            # Replace the line 'go 1.22' with 'go 1.22.0' for example
+        gomod_path = dest_dir.joinpath('go.mod')
+        if gomod_path.exists():
+            # Read the gomod contents
+            new_lines = []
+            with open(gomod_path, "r") as file:
+                lines = file.readlines()
+                for line in lines:
+                    stripped_line = line.strip()
+                    match = re.match(r"(^go \d\.\d+$)", stripped_line)
+                    if match:
+                        # Append a .0 to the go mod version, if it exists
+                        # Replace the line 'go 1.22' with 'go 1.22.0' for example
+                        go_version_string = match.group(1)  # eg. 'go 1.23'
+                        go_version_number = float(go_version_string.split(" ")[-1])  # eg. 1.23
```

Comment:
Converting version numbers to float looks error-prone to me. Is it possible to convert that to a tuple of integer or use a library like `semver`?

Reply:
thanks yuxiang, added in https://github.com/openshift-eng/art-tools/pull/1718/commits/e2e51ff923bd035ec2622cff344db4164cfa9d89